### DISCUSSION TITLE
OpenRadioss，2D Fluid-structure blast

### DISCUSSION THREAD

**Discussion by HNUbigbird** [OpenRadioss/OpenRadioss#2554] [Q&A]
I would like to use OpenRadioss for a fluid-structure interaction explosion analysis. The model is a 2D axisymmetric model, and the commonly used TYPE18 contact seems to only apply to three-dimensional problems. Which type of contact should I use to simulate fluid-structure interaction explosion?

**Comment by wanglei742** [comment]
Hi, 
I came across the same question about way of modeling 2d solid fluid interaction, as my understanding, type18 is not supported the 2d model,so you can use the multimaterial law and along the sharing the node between parts to achieve your goal.

**Reply by HNUbigbird** [reply]
Hi，
Thank you very much for your reply. If I follow the shared node approach, should the mesh sizes for the fluid and solid be the same? In explosions, the mesh for fluids needs to be very small, while the mesh for solids is set larger. How can this issue be resolved?

**Reply by wanglei742** [reply]
Hi, 
you have to trade off the precision and time. maybe you can use a relative coarse mesh at start, then refine the solid part mesh size, and study the impacts.
I found an example in altair community, even this is not an axis-symm example, hope this can help in some extend
https://community.altair.com/community/en/mok-fluid-structure-interaction-fsi-benchmark-simulation-with-radioss?id=kb_article&sys_id=67a8910497554610e3b0361e6253afd7

**Reply by HNUbigbird** [reply]
thanks，you help me a lot.

**Comment by RevolT87** [comment]
Hi @HNUbigbird,

I confirm that type18 interface is not compatible with 2D analysis. You may build a 1-layer 3D model if you really want/need to use /INTER/TYPE18. The example shared by @wanglei742 is good to ilustrate this. 

You may try to use /INTER/TYPE1: ALE boundary nodes can be tied to Lagrangian surface to make fluid structure interaction. This way Lagrangian surface is acting as a sliding wall boundary condition.

Depending on what kind of explosion simulation you want to do, you may also consider mapping. You do a first pure euler simulation in 2D and then you map the results in a 3D CEL model. This is what is described here: https://community.altair.com/community?id=altair_file_exchange_article_view&sysparm_article=KB0119695 

Regards,
Mathis.

**Reply by HNUbigbird** [reply]
Hi，
Thank you very much for your reply. I tried using CONTACT TYPE1 with the explosive grid as the secondary node and the rock medium as the main node, and encountered the following error: TIED INTERFACE: SECONDARY NODE(S) OUTSIDE MAIN SEGMENT.
What should I do ？I would appreciate it if you can reply me.
![blast](https://github.com/OpenRadioss/OpenRadioss/assets/167994304/cf55e31d-7794-45e4-a9d8-19b22d2b0469)

**Reply by wanglei742** [reply]
Hi, 
I thought this is the reason that type1 is bonding the lagrangian and fluid part, but for your problem both parts are fluid(explsosive TNT and air/water), so it's not fit for your problem.

**Reply by HNUbigbird** [reply]
Hi，
My explosive TNT is the fluid part, and the surrounding medium is rock (Lagrangian algorithm), but using TYPE1 will result in an error.

**Comment by hnubiggestbird** [comment]
Oh, I think your bird is not big
